src/main/java/com/google/sps/servlets/ChatroomServlet.java

```diff
@@ -50,18 +50,17 @@
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-        String userID = request.getParameter("userId");
-
+        String userID = request.getHeader("userId");
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-        if (userID == null) {
+        if (userID == null || recipientID == null) {
```

Comment:
same here. i think this error check should be handled in authfilter so that we dont have to do this in every other servlet??

Reply:
^^. Outdated as well